### PR TITLE
Cirrus: Initial implementation

### PR DESCRIPTION
**Pull request by cevich**
Based on containers/storage

Signed-off-by: Chris Evich <cevich@redhat.com>

### CODE REVIEW

**Comment by cevich** [comment]
Failure is expected.

**Comment by cevich** [comment]
okay, this is minimally functional now.  Though I haven't tested `hack/get_ci_vm.sh` yet.

**Comment by rhatdan** [comment]
LGTM
@nalind @TomSweeneyRedHat @giuseppe @vrothberg PTAL

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+# Default timeout for each task
+timeout_in: 120m
+
+# Default VM to use unless set or modified by task
+gce_instance:
+    image_project: "${IMAGE_PROJECT}"
+    zone: "us-central1-c"  # Required by Cirrus for the time being
+    cpu: 2
+    memory: "4Gb"
```
Comment (TomSweeneyRedHat) [inline .cirrus.yml]: Is this the same level as Podman?  I'd have thought 6Gb would have been the number here.  But if this looks sufficient, I'm fine with it as is.

File: .cirrus.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    disk: 200
```
Comment (TomSweeneyRedHat) [inline .cirrus.yml]: What's 200?  If it doesn't requre Mb or Gb as a qualifier, might be nice to add a quick comment.

File: .cirrus.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        failure_rh_audit_log_script: '${_RAUDITCMD} || true'
+        failure_ubuntu_audit_log_script: '${_UAUDITCMD} || true'
+        failure_journal_log_script: '${_JOURNALCMD} || true'
+
+
+# Update metadata on VM images referenced by this repository state
+meta_task:
+
+    container:
+        image: "quay.io/libpod/imgts:latest"  # see contrib/imgts
```
Comment (TomSweeneyRedHat) [inline .cirrus.yml]: Knowledge question for me.  Where/how does this image get created?

File: contrib/cirrus/setup.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    glib2-devel
+    go-md2man
+    golang
+    gpgme-devel
+    libassuan-devel
+    libseccomp-devel
+    ostree-devel
+    podman
+    python2
+    python3
```
Comment (TomSweeneyRedHat) [inline contrib/cirrus/setup.sh]: Do we really need both python2 and 3

File: contrib/cirrus/setup.sh [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    libostree-dev
+    bats
+    aufs-tools
+"
+
+FEDORA_PACKAGES="
+    bats
+    btrfs-progs-devel
+    bzip2
+    curl
```
Comment (TomSweeneyRedHat) [inline contrib/cirrus/setup.sh]: Don't we need containers-common?

File: contrib/cirrus/setup.sh [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    golang
+    gpgme-devel
+    libassuan-devel
+    libseccomp-devel
+    ostree-devel
+    podman
+    python2
+    python3
+    redhat-rpm-config
+    rsync
```
Comment (TomSweeneyRedHat) [inline contrib/cirrus/setup.sh]: Is runc getting chunked in from somewhere else?

File: contrib/cirrus/setup.sh [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    golang
+    go-md2man
+    libgpgme11-dev
+    libdevmapper-dev
+    libseccomp-dev
+    libglib2.0-dev
+    libostree-dev
+    bats
+    aufs-tools
+"
```
Comment (TomSweeneyRedHat) [inline contrib/cirrus/setup.sh]: Think we need btrfs-tools libpapparmor-dev libselinux-dev skopeo-containers (Maybe containers-common now) .
Would be nice to have these alphabetized as you have the ones in Fedora

File: hack/get_ci_vm.sh [hunk 8]
```diff
@@ -0,0 +1,215 @@
+#!/bin/bash
+
+set -e
+
+RED="\e[1;36;41m"
+YEL="\e[1;33;44m"
+NOR="\e[0m"
+USAGE_WARNING="
+${YEL}WARNING: This will not work without local sudo access to run podman,${NOR}
```
Comment (TomSweeneyRedHat) [inline hack/get_ci_vm.sh]: podman or buildah here?

File: .cirrus.yml (hunk 3 above)
Comment (TomSweeneyRedHat) [reply .cirrus.yml]: Is that part of this test process?

**Comment by TomSweeneyRedHat** [comment]
Nice start overall @cevich Just a few nits and a question.

**Comment by cevich** [comment]
okay, tested `hack/get_ci_vm.sh` seems to work for me, though would be good to confirm it works for @nalind

**Comment by cevich** [comment]
@TomSweeneyRedHat thanks a bunch for your review, going through it now.

File: .cirrus.yml (hunk 1 above)
Comment (cevich) [reply .cirrus.yml]: yep, copy-paste from over there.  CPU, Memory and number of VMs are the main cost items, but I have no problem increasing them if they prove to be insufficient.

File: .cirrus.yml (hunk 3 above)
Comment (cevich) [reply .cirrus.yml]: Brent made a 'libpod' user on quay, and a 'imgts' image repository.  It's configured to track upstream 'containers/libpod' (master) for changes, and rebuild the container if so.  Fairly simple/standard setup, but happy to show if you need more in-depth.  Ask Brent if you want a repository setup for something under that `libpod` namespace.

File: .cirrus.yml (hunk 2 above)
Comment (cevich) [reply .cirrus.yml]: It's Gb, and it's 200 for a reason, so yes it does need a comment, thanks.

File: contrib/cirrus/setup.sh (hunk 4 above)
Comment (cevich) [reply contrib/cirrus/setup.sh]: No idea at all TBH, this was mostly just copy-paste from somewhere else.

I'll add a comment, mentioning these lists needs to be tweaked.

File: contrib/cirrus/setup.sh (hunk 5 above)
Comment (cevich) [reply contrib/cirrus/setup.sh]: okay :smile:

File: contrib/cirrus/setup.sh (hunk 4 above)
Comment (rhatdan) [reply contrib/cirrus/setup.sh]: I don't think we need any python.

File: contrib/cirrus/setup.sh (hunk 5 above)
Comment (rhatdan) [reply contrib/cirrus/setup.sh]: Installing podman will pull it,

File: contrib/cirrus/setup.sh (hunk 6 above)
Comment (cevich) [reply contrib/cirrus/setup.sh]: Nope, it's completely dead-simple WYSIWYG here :smiley: I'll add runc.

Comment (rhatdan) [reply contrib/cirrus/setup.sh]: podman install pulls it in.

File: contrib/cirrus/setup.sh (hunk 7 above)
Comment (cevich) [reply contrib/cirrus/setup.sh]: Added, and sorted.

File: hack/get_ci_vm.sh (hunk 8 above)
Comment (cevich) [reply hack/get_ci_vm.sh]: Good catch, nope, `podman` for sure.  It uses a container image I maintain that includes google's cloud tooling and some magic to not mess up file permissions on the host while running.

File: contrib/cirrus/setup.sh (hunk 7 above)
Comment (cevich) [reply contrib/cirrus/setup.sh]: Actually adding those failed.  I'm going to leave sorting out the packaging as a TODO for when the testing is being implemented.  Added a comment to that effect.

**Comment by cevich** [comment]
@TomSweeneyRedHat @rhatdan @nalind PTAL I think this is good'nuff for now.  Though I would like someone to confirm `hack/get_ci_vm.sh` is working in this repo., I only gave it a quick poke and I've got so many tools and configs installed locally, it may have worked by accident.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**